scaleway/resource_iam_group.go

```diff
@@ -0,0 +1,135 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "The name of the iam group",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the iam group",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the group",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the group",
+			},
+			"user_ids": {
+				Type:        schema.TypeList,
+				Description: "List of IDs of the users attached to the group",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validationUUID(),
+				},
+			},
+			"application_ids": {
+				Type:        schema.TypeList,
+				Description: "List of IDs of the applications attached to the group",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validationUUID(),
+				},
+			},
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayIamGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := iamAPI(meta)
+	group, err := api.CreateGroup(&iam.CreateGroupRequest{
+		OrganizationID: d.Get("organization_id").(string),
+		Name:           expandOrGenerateString(d.Get("name"), "group-"),
+		Description:    d.Get("description").(string),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(group.ID)
+
+	return resourceScalewayIamGroupRead(ctx, d, meta)
+}
+
+func resourceScalewayIamGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := iamAPI(meta)
+	group, err := api.GetGroup(&iam.GetGroupRequest{
+		GroupID: d.Id(),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_ = d.Set("name", group.Name)
+	_ = d.Set("description", group.Description)
+	_ = d.Set("created_at", flattenTime(group.CreatedAt))
+	_ = d.Set("updated_at", flattenTime(group.UpdatedAt))
+	_ = d.Set("organization_id", group.OrganizationID)
+	_ = d.Set("user_ids", group.UserIDs)
+	_ = d.Set("application_ids", group.ApplicationIDs)
+
+	return nil
+}
+
+func resourceScalewayIamGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := iamAPI(meta)
+
+	req := &iam.UpdateGroupRequest{
+		GroupID: d.Id(),
+	}
+
+	if d.HasChange("name") {
+		req.Name = expandStringPtr(d.Get("name"))
+	}
+	if d.HasChange("description") {
+		req.Description = expandStringPtr(d.Get("description"))
+	}
+
+	_, err := api.UpdateGroup(req, scw.WithContext(ctx))
```

Comment:
Please verify if you set only one of both we keep the attribute that does not change.